fastapi/openapi/utils.py

```diff
@@ -437,3 +453,123 @@ def get_openapi(
     if tags:
         output["tags"] = tags
     return jsonable_encoder(OpenAPI(**output), by_alias=True, exclude_none=True)  # type: ignore
+
+
+def merge_paths(
+    existing_path: Dict[str, Any], new_path: Dict[str, Any]
+) -> Dict[str, Any]:
+    """Merge two openapi path descriptions for the same route, e.g. for response content with different accept-encoding."""
+    path_by_operation_id: Dict[str, Dict[str, Any]] = {}
+    for method, operation in itertools.chain(existing_path.items(), new_path.items()):
+        path_by_operation_id.setdefault(operation["operationId"], {})[
+            method
+        ] = operation
+    return deep_dict_operation_merge(path_by_operation_id) or {}
+
+
+def merge_tags(tags_by_id: Dict[str, Optional[List[str]]]) -> Optional[List[str]]:
+    tags: Optional[List[str]] = None
+    for update_tags in tags_by_id.values():
+        if update_tags:
+            tags = tags or []
+            tags.extend(t for t in update_tags if t not in tags)
+    return tags
+
+
+def merge_summary(summary_by_id: Dict[str, Optional[str]]) -> Optional[str]:
+    summary: Optional[str] = None
+    for update_summary in summary_by_id.values():
+        if update_summary and summary != update_summary:
+            summary = f"{summary} / {update_summary}" if summary else update_summary
+    return summary
+
+
+def merge_description(desc_by_id: Dict[str, Optional[str]]) -> Optional[str]:
+    desc: Optional[str] = None
+    for update_id, update_desc in desc_by_id.items():
```

Comment:
Since you're not accessing update_id you could do
```suggestion
    for update_desc in desc_by_id.values():
```